iceoryx_posh/source/roudi/roudi.cpp

```diff
@@ -231,7 +231,16 @@ void RouDi::processMessage(const runtime::IpcMessage& message,
         }
         else
         {
-            capro::ServiceDescription service(cxx::Serialization(message.getElementAtIndex(2)));
+            auto maybeService =
```

Comment:
It's a long time since we hacked on iceoryx so correct me if I'm wrong but didn't we use `maybeFoo` for `optional<Foo> bar()` and `fooResult` for `expected<Bar> foo()`?

Reply:
I don't quite know, if we agreed on this before, but I like the idea a lot! Adapted the code and added a hint in `CONTRIBUTING.md`.